src/thorlabs_cube/driver/tcube/tsc.py

```diff
@@ -0,0 +1,407 @@
+        )  # FIXME->need to change serialization format but should work after that
+
+    def is_channel_enabled(self):
+        return super().is_channel_enabled()
+
+    def ping(self):
+        return super().ping()
+
+    async def set_channel_enable_state(self, activated):
+        return await super().set_channel_enable_state(activated)
+
+    async def get_channel_enable_state(self):
+        return await super().get_channel_enable_state()
+
+    # functions from base class definition end
+
+    async def get_bay_used(self):
+        """Identify which bay is being used by the controller on Thorlabs Hub
+
+        :param absolute_position: The absolute position in encoder counts.
+                                E.g., 200,000 counts for 10 mm.
+        """
+        get_msg = await self.send_request(
+            MGMSG.HUB_REQ_BAYUSED, [MGMSG.HUB_GET_BAYUSED], 1
+        )
+
+        used_bay_ident = get_msg.param1
+        return used_bay_ident
+
+    async def set_absolute_position(self, absolute_position):
+        """Move the motor to an absolute position.
+
+        :param absolute_position: The absolute position in encoder counts.
+                                E.g., 200,000 counts for 10 mm.
+        """
+
+        # Pack the Channel ID (2 bytes) and the Absolute Position (4 bytes)
+        payload = st.pack("<Hl", 1, absolute_position)
+
+        await self.send_request(
+            MGMSG.MOT_MOVE_ABSOLUTE,
+            [MGMSG.MOT_MOVE_COMPLETED, MGMSG.MOT_MOVE_STOPPED],
+            data=payload,
+        )
+
+    async def move_stop(self, stop_mode):
+        """Stop any type of motor move (relative, absolute, homing, or velocity).
+
+        :param stop_mode: The stop mode (1 for immediate stop, 2 for profiled stop).
+        """
+
+        # Send the message with the command ID (0x0465), and put channel ID in param1 and stop mode in param2
+        await self.send_request(
+            "<BB",
+            MGMSG.MOT_MOVE_STOP,
+            [MGMSG.MOT_MOVE_STOPPED],
+            param1=_CHANNEL,
+            param2=stop_mode,
+        )
+
+    async def set_av_modes(self, mode_bits):
+        """Set the LED indicator modes based on the provided mode_bits.
+
+        :param mode_bits: A bitmask indicating which modes to enable:
+                        - 1 (LEDMODE_IDENT): LED flashes when 'Ident' is sent.
+                        - 2 (LEDMODE_LIMITSWITCH): LED flashes when motor reaches limit switch.
+                        - 8 (LEDMODE_MOVING): LED is lit when the motor is moving.
+        """
+
+        # Pack the Channel ID (2 bytes) and ModeBits (4 bytes)
+        payload = st.pack("<Hl", 1, mode_bits)
+
+        # Send the message with the packed payload for MOT_SET_AVMODES (0x04B3)
+        await self.send(Message(MGMSG.MOT_SET_AVMODES, data=payload))
+
+    async def get_av_modes(self):
+        """Get the current LED indicator mode bits.
+
+        :return: The LED mode bits.
```

Comment:
More information is required here. How should the user interpret the bits? Explain here or link to `set_av_modes`, where it is explained.